docs: rework Browser chapter (#503)
oasisprotocol/sapphire-paratime: docs/develop/browser.md

---
description: Integrating Sapphire in your web dApp
---

# Browser Support

This page provides guidance for developers looking to build confidential dApps

> on Sapphire that work across different web browsers and integrates with

Reviewer: ```suggestion
on Sapphire that work across different web browsers and integrate with
```

wallets, including Metamask. It covers supported libraries, best practices for
secure transactions, and quick steps for using the libraries.

## Supported Libraries

> | Library                                       | Package                                           | API Reference     | GitHub                  |

Reviewer: ```suggestion
| Library                                       | Package                                           | API Reference     | Source                  |
```

use our library-specific packages for [Ethers][ethers-quick],
[Viem][viem-quick] or [Wagmi][wagmi-quick].

[s-p]: ./browser.md#lightweight-sapphire-typescript-wrapper
[ethers-quick]: ./browser.md#ethers-v6
[viem-quick]: ./browser.md#viem
[wagmi-quick]: ./browser.md#wagmi

:::info Example: Starter project

If your project includes both a smart contract backend and a web frontend, you
can explore our **[demo-starter]** repository. It provides a working example using
React as well as a [Vue branch].

:::

[demo-starter]: https://github.com/oasisprotocol/demo-starter
[Vue branch]: https://github.com/oasisprotocol/demo-starter/tree/vue

## Transaction encryption

When using the supported libraries, ensure that all transactions containing
sensitive information **are encrypted**. Encryption is essential to safeguard user
data and ensure privacy. To verify that a transaction is encrypted, you can
check the transaction details on the Oasis Block Explorer for the corresponding
network ([Localnet], [Testnet], or [Mainnet]). Look for a green lock icon next
to the transaction, which indicates that it is securely encrypted.

:::tip Check Calldata Encryption Programmatically

> You can check programmatically if calldata is encrypted by using the
> `isCalldataEnveloped` function of the `@oasisprotocol/sapphire-paratime`
> package.

Reviewer: ```suggestion
You can check programmatically if calldata is encrypted by using
[`isCalldataEnveloped()`].

[`isCalldataEnveloped()`]: https://api.docs.oasis.io/js/sapphire-paratime/functions/isCalldataEnveloped.html
```
